packages/typescript-estree/src/convert.ts

```diff
@@ -2546,19 +2546,32 @@ export class Converter {
         return result;
       }
 
-      case SyntaxKind.ImportType:
-        return this.createNode<TSESTree.TSImportType>(node, {
+      case SyntaxKind.ImportType: {
+        const range = getRange(node, this.ast);
+        if (node.isTypeOf) {
+          const token = findNextToken(node.getFirstToken()!, node, this.ast)!;
+          range[0] = token.getStart(this.ast);
+        }
+        const result = this.createNode<TSESTree.TSImportType>(node, {
           type: AST_NODE_TYPES.TSImportType,
-          isTypeOf: !!node.isTypeOf,
-          parameter: this.convertChild(node.argument),
+          argument: this.convertChild(node.argument),
```

Comment:
i renamed this property from parameter to argument as is more aligned with typescript/babel and rest of types

eg, call expression has arguments

Reply:
alternativelly we could align this node with import expression and use source